xapian-letor/feature/colltfcolllenfeature.cc

```diff
@@ -38,37 +38,74 @@ CollTfCollLenFeature::name() const
 vector<double>
 CollTfCollLenFeature::get_values() const
 {
-    LOGCALL(API, std::vector<double>, "CollTfCollLenFeature::get_values", NO_ARGS);
+    LOGCALL(API, vector<double>, "CollTfCollLenFeature::get_values", NO_ARGS);
 
     vector<double> values;
     double value = 0;
+    double coll_len = 0;
+    map<string, Xapian::termcount>::const_iterator coll_len_iterator =
+	    collection_length.find("title");
+    if (coll_len_iterator != collection_length.end())
+	coll_len = (double)coll_len_iterator->second;
+    else
+	coll_len = 0;
 
     for (Xapian::TermIterator qt = feature_query.get_unique_terms_begin();
 	 qt != feature_query.get_terms_end(); ++qt) {
-	if ((*qt).substr(0, 1) == "S" || (*qt).substr(1, 1) == "S")
-	    value += log10(1 + ((double)collection_length.find("title")->second
-				/ (double)(1 + collection_termfreq.find(*qt)->second)));
+	if ((*qt).substr(0, 1) == "S" || (*qt).substr(1, 1) == "S") {
+	    map<string, Xapian::termcount>::const_iterator coll_tf_iterator =
+		    collection_termfreq.find(*qt);
+	    if (coll_tf_iterator != collection_termfreq.end()) {
+		value += log10(1 + (coll_len /
+				    (double)(1 + coll_tf_iterator->second)));
+	    }
+	    else
```

Comment:
Xapian coding style is that you don't put a newline between `}` and `else`.

It's also better to put `{` and `}` around both or neither of the blocks.

This issue is repeated below.